src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/GraphvizExporter.java

```diff
@@ -0,0 +1,151 @@
+    public void exportDiffTree(DiffTree diffTree, OutputStream destination) throws IOException {
+        var output = new PrintStream(destination);
+
+        output.println("digraph g {");
+
+        format.forEachNode(diffTree, (node) -> {
+            output.format("  %d [label=\"%s\"];%n",
+                    node.getID(),
+                    escape(format.getNodeFormat().toMultilineLabel(node)));
+        });
+
+        format.forEachEdge(diffTree, (edge) -> {
+            output.format("  %d -> %d;%n", edge.from().getID(), edge.to().getID());
+        });
+
+        output.println("}");
+        output.flush();
+    }
+
+    /**
+     * Runs the Graphviz {@code dot} program returning its result.
+     *
+     * @param diffTree is the tree to be layouted by Graphviz.
+     * @param outputFormat is the requested format which is passed to the {@code dot} program with
+     * the {@code -T} flag.
+     * @return a buffered {@code InputStream} of the Graphviz output
+     */
+    public InputStream computeGraphvizLayout(
+            DiffTree diffTree,
+            LayoutAlgorithm algorithm,
+            OutputFormat outputFormat)
+            throws IOException {
+        // Print error messages to stderr so grogramming errors in {@code exportDiffTree} can be
+        // diagnosed more easily.
+        var graphvizProcess =
+            new ProcessBuilder(algorithm.getExecutableName(), "-T" + outputFormat.getFormatName())
+            .redirectInput(ProcessBuilder.Redirect.PIPE)
+            .redirectOutput(ProcessBuilder.Redirect.PIPE)
+            .redirectError(ProcessBuilder.Redirect.INHERIT)
+            .start();
+
+        // This could lead to a dead lock if {@code graphvizProcess} is not consuming its input and
+        // the OS buffer fills up, but Graphviz needs the whole graph to generate a layout so this
+        // should be safe.
+        var graphizInput = new BufferedOutputStream(graphvizProcess.getOutputStream());
+        exportDiffTree(diffTree, graphizInput);
+        graphizInput.close();
+
+        return new BufferedInputStream(graphvizProcess.getInputStream());
+    }
+
+    /**
+     * Replaces all special characters with uninterpreted escape codes.
+     *
+     * The Graphviz parser for strings interprets some characters specially. The result of this
+     * function can be used in Graphviz strings (surrounded by double quotes) resulting in all
+     * characters appearing literally in the output.
+     *
+     * Note that some backends of Graphviz may still interpret some strings specially, most
+     * commonly strings containing HTML tags.
+     *
+     * @param label a list of lines to be used as verbatim label
+     * @return a single string which produces the lines of {@code label} verbatim
+     */
+    static private String escape(List<String> label) {
+        return label
+            .stream()
+            .map((line) -> quotePattern.matcher(line).replaceAll((match) -> "\\\\" + match.group()))
```

Comment:
Turns out that this escaping code is buggy (e.g. if `match.group() == "\\"`), but I already fixed this locally.

Reply:
Fixed in #54